openfl-workspace/torch_cnn_mnist/README.md

```diff
@@ -0,0 +1,212 @@
+
+To help OpenFL calculate the initial model weights, we need to provide the shape of the input tensor as an additional parameter. For the MNIST data set of grayscale (single-channel) 28x28 pixel images, the input tensor shape is [1,28,28]. We will also use a locally deployed aggregator (localhost). Thus, the workspace initialization command for our local federation becomes:
+
+```
+mkdir save
+fx plan initialize --input_shape [1,28,28] --aggregator_address localhost
+```
+
+We can now perform a test run with the following commands for creating a local PKI setup and starting the aggregator and the collaborators on the same machine:
+
+```
+cd ~/openfl/openfl-tutorials/taskrunner/
+
+# This will create a local certificate authority (CA), so the participants communicate over a secure TLS Channel
+fx workspace certify
+
+#################################################################
+# Step 1: Setup the Aggregator #
+#################################################################
+
+# Generate a Certificate Signing Request (CSR) for the Aggregator
+fx aggregator generate-cert-request --fqdn localhost
+
+# The CA signs the aggregator's request, which is now available in the workspace
+fx aggregator certify --fqdn localhost --silent
+
+################################
+# Step 2: Setup Collaborator 1 #
+################################
+
+# Create a collaborator named "collaborator1" that will use data path "data/1"
+# This command adds the collaborator1,data/1 entry in data.yaml
+fx collaborator create -n collaborator1 -d 1
+
+# Generate a CSR for collaborator1
+fx collaborator generate-cert-request -n collaborator1
+
+# The CA signs collaborator1's certificate, adding an entry to the authorized cols.yaml
+fx collaborator certify -n collaborator1 --silent
+
+################################
+# Step 3: Setup Collaborator 2 #
+################################
+
+# Create a collaborator named "collaborator2" that will use data path "data/2"
+# This command adds the collaborator2,data/2 entry in data.yaml
+fx collaborator create -n collaborator2 -d 2
+
+# Generate a CSR for collaborator2
+fx collaborator generate-cert-request -n collaborator2
+
+# The CA signs collaborator2's certificate, adding an entry to the authorized cols.yaml
+fx collaborator certify -n collaborator2 --silent
+
+##############################
+# Step 4. Run the Federation #
+##############################
+
+fx aggregator start & fx collaborator start -n collaborator1 & fx collaborator start -n collaborator2
+```
+
+A successful local simulation of the FL workspace involves the aggregator and collaborators completing a round of training, saving the best-performing model under save/best.pbuf, and exiting with a unanimous “End of Federation reached…”:
+
+## output
```

Comment:
nit - Sample output

Reply:
resolved in be37d1c1e35fee2cce4ee8907547c861e28e6d8f